libraries/abstractions/common_io/include/iot_spi.h

```diff
@@ -155,86 +163,132 @@ void iot_spi_set_callback( IotSPIHandle_t const pxSPIPeripheral,
 /**
  * @brief Configures the SPI port with user configuration.
  *
- * The application should call this function to configure the SPI driver
- * with values in the structure IotSPIMasterConfig_t. The application also needs to
- * specify the request which is as per enum IotSPIIoctlRequest_t.
+ *
+ * @note eSPISetMasterConfig is expected only called once at beginning.
+ * This request expects the buffer with size of IotSPIMasterConfig_t.
+ *
+ * @note eSPIGetMasterConfig gets the current configuration for SPI master.
+ * This request expects the buffer with size of IotSPIMasterConfig_t.
+ *
+ * @note eSPIGetTxNoOfbytes returns the number of written bytes in last operation.
+ * This is supposed to be called in the caller task or application callback, right after last operation completes.
+ * This request expects 2 bytes buffer (uint16_t).
+ *
+ * - If the last operation only did write, this returns the actual number of written bytes which might be smaller than the requested number (partial write).
+ * - If the last operation only did read, this returns 0.
+ * - If the last operation did both write and read, this returns the number of written bytes.
+ *
+ * @note eSPIGetRxNoOfbytes returns the number of read bytes in last operation.
+ * This is supposed to be called in the caller task or application callback, right after last operation completes.
+ * This request expects 2 bytes buffer (uint16_t).
+ *
+ * - If the last operation only did read, this returns the actual number of read bytes which might be smaller than the requested number (partial read).
+ * - If the last operation only did write, this returns 0.
+ * - If the last operation did both write and read, this returns the number of read bytes.
  *
  * @param[in] pxSPIPeripheral The SPI peripheral handle returned in open() call.
  * @param[in] xSPIRequest The configuration request from one of the
  * IotSPIIoctlRequest_t.
  * @param[in,out] pvBuffer The configuration values for the SPI port.
  *
  *
- * @return SUCCESS = IOT_SPI_SUCESS,
- *         FAILED = IOT_SPI_FUNCTION_NOT_SUPPORTED,
- *         SPI_BUSY = IOT_SPI_BUS_BUSY,
- *          Master_UNINITIALIZED = IOT_SPI_MASTER_NOT_INITIALIZED
+ * @return
+ * - IOT_SPI_SUCCESS, on success
+ * - IOT_SPI_INVALID_VALUE, if
+ *     - pxSPIPeripheral is NULL
+ *     - pxSPIPeripheral is not opened yet
+ *     - pucBuffer is NULL with requests which needs buffer
  */
 int32_t iot_spi_ioctl( IotSPIHandle_t const pxSPIPeripheral,
                        IotSPIIoctlRequest_t xSPIRequest,
                        void * const pvBuffer );
 
 /**
- * @brief The SPI master starts reading the number of bytes from the slave
- * synchronously.
+ * @brief The SPI master starts reading from the slave synchronously.
+ *
+ * This function attempts to read certain number of bytes from slave device to a pre-allocated buffer, in synchronous way.
+ * Partial read might happen.
```

Comment:
nit: "Partial read might happen", -> Can we be clear here please? the API does not return unless there is an error even if there is a partial read

Reply:
Good point.